overlord/snapstate/handlers.go

```diff
@@ -818,6 +852,20 @@ func (m *SnapManager) undoUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// Restore disabled services
+	var missingSvcs []string
+	missingSvcs, err = m.backend.RestoreDisabledServices(oldInfo, snapst.LastActiveDisabledServices, progress.Null)
```

Comment:
I think other places where we run external commands usually execute with state being unlocked. You may need to do 

```
st.Unlock()
missingSvcs, err = m.backend.RestoreDisabledServices(oldInfo, snapst.LastActiveDisabledServices, progress.Null)
st.Lock()
if err != nil {
    t.Errorf("cannot undo unlinking of current snap, failed to disable services: %v", err)
    return err
}
```

Reply:
Okay, unlocked the state around everywhere that RestoreDisabledServices is called